File: api/app/Support/DocumentTotals.php
<?php
// app/Support/DocumentTotals.php

namespace App\Support;

class DocumentTotals
{
    /**
     * Kalemlerden ara toplam, KDV ve genel toplamı hesaplar.
     * items: [{quantity, unit_price, ...}]
     */
    public static function compute(array $items, float $taxRate = 20, float $discount = 0): array
    {
        $subtotal = 0;
        foreach ($items as $item) {
            $qty   = (float) ($item['quantity'] ?? 1);
            $price = (float) ($item['unit_price'] ?? 0);
            $subtotal += $qty * $price;
        }

        $taxAmount = round(($subtotal - $discount) * $taxRate / 100, 2);
        $total = round($subtotal - $discount + $taxAmount, 2);

        return [
            'subtotal'   => round($subtotal, 2),
            'tax_rate'   => $taxRate,
            'tax_amount' => $taxAmount,
            'discount'   => round($discount, 2),
            'total'      => $total,
        ];
    }
}

File: api-net/Services/DocumentTotals.cs
namespace LiftOtonom.Api.Services;

public record LineItem(string? Description, decimal? Quantity, decimal? UnitPrice);

public static class DocumentTotals
{
    public record Totals(decimal Subtotal, decimal TaxRate, decimal TaxAmount, decimal Discount, decimal Total);

    public static Totals Compute(IEnumerable<LineItem>? items, decimal taxRate = 20, decimal discount = 0)
    {
        decimal subtotal = 0;
        if (items != null)
            foreach (var i in items)
                subtotal += (i.Quantity ?? 1) * (i.UnitPrice ?? 0);

        var taxAmount = Math.Round((subtotal - discount) * taxRate / 100, 2);
        var total = Math.Round(subtotal - discount + taxAmount, 2);
        return new Totals(Math.Round(subtotal, 2), taxRate, taxAmount, Math.Round(discount, 2), total);
    }
}
